src/analysis/processing/qgsalgorithmextractlabels.cpp

```diff
@@ -377,10 +389,12 @@ QVariantMap QgsExtractLabelsAlgorithm::processAlgorithm( const QVariantMap &para
     textFormat.setColor( QColor( 0, 0, 0 ) );
     settings.setFormat( textFormat );
 
-    settingsProperties.setProperty( QgsPalLayerSettings::Color, QgsProperty::fromExpression( QStringLiteral( "if(\"LabelUnplaced\",'255,0,0','0,0,0')" ) ) );
+    settingsProperties.setProperty( QgsPalLayerSettings::Color, QgsProperty::fromExpression( QStringLiteral( "if(\"LabelUnplaced\",'255,0,0',\"Color\")" ) ) );
```

Comment:
These should all be QgsProperty::fromField( ... )

Reply:
@nyalldawson , done.